gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -293,19 +294,27 @@ public AddSpecResponse onAddSpec(Spec addedSpec) {
           response = Arrays.toString(flowSpec.getCompilationErrors().toArray());
         }
 
-        if (!isExplain) {
+        boolean compileSuccess = FlowCatalog.isCompileSuccessful(response);
+
+        if (!isExplain && compileSuccess) {
           this.scheduledFlowSpecs.put(addedSpec.getUri().toString(), addedSpec);
 
           if (jobConfig.containsKey(ConfigurationKeys.JOB_SCHEDULE_KEY)) {
             _log.info("{} Scheduling flow spec: {} ", this.serviceName, addedSpec);
             scheduleJob(jobConfig, null);
             if (PropertiesUtils.getPropAsBoolean(jobConfig, ConfigurationKeys.FLOW_RUN_IMMEDIATELY, "false")) {
               _log.info("RunImmediately requested, hence executing FlowSpec: " + addedSpec);
-              this.jobExecutor.execute(new NonScheduledJobRunner(flowSpec.getUri(), false, jobConfig, null));
+              this.jobExecutor.execute(new NonScheduledJobRunner(flowSpecUri, false, jobConfig, null));
             }
           } else {
             _log.info("No FlowSpec schedule found, so running FlowSpec: " + addedSpec);
-            this.jobExecutor.execute(new NonScheduledJobRunner(flowSpec.getUri(), true, jobConfig, null));
+            this.jobExecutor.execute(new NonScheduledJobRunner(flowSpecUri, true, jobConfig, null));
+          }
+        } else {
+          _log.info("Removing the flow spec: {}, since it is an EXPLAIN request or the flow compilation failed.", addedSpec);
```

Comment:
Any chance we can distinguish the log message based on whether it is an explain query of compile failed? Will help debugging.

Reply:
updated